File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;
class Solution {
    public int solution(int[][] routes) {
        int answer = 0;
        Arrays.sort(routes, new Comparator<int[]>(){
            @Override
            public int compare(int[] a, int[] b){
                return a[1] != b[1] ? a[1] - b[1] : a[0] - b[0];
            }
        });
        int cam_pos = Integer.MIN_VALUE;
        for(int[] info : routes){
            if(info[0] > cam_pos){
                cam_pos = info[1];
                answer += 1;
            }
        }
        return answer;
    }
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.js
function solution(routes) {
  var answer = 0;
  var arr = routes.sort((a, b) => (a[1] === b[1] ? a[0] - b[0] : a[1] - b[1]));
  var camera_pos = -30001; // 이 이상 안 준다고 하니깐..
  for (var i = 0; i < arr.length; i++) {
    if (camera_pos < arr[i][0]) {
      answer += 1;
      camera_pos = arr[i][1];
    }
  }
  return answer;
}